### PR TITLE
Update synapse deploy

### LINKED ISSUE #204: Update Synapse Deploy Logic
**Description**

Currently Synapse notebooks are only created by the Starter Kit deployment process if they do not already exist. Unfortunately this means that whenever we make changes to an existing notebook we must delete it from Synapse in order for the order version to be deployed by our CD pipeline on the next deployment. We need to modify this process so that the CD pipeline is sophisticated enough to handle updating notebooks, not just creating them if they don't already exist.

**Acceptance Criteria**
GIVEN the Starter Kit Deployment workflow
WHEN the workflow is run
THEN any changes to the Synapse notebooks are successfully deployed to Azure regardless of whether a notebook by the same name already exists.

### PR DESCRIPTION
**Pull request by DanPaseltiner**
This PR updates our deployment workflow to update our Synapse notebooks, pipeline, and trigger whenever changes to those areas of the repo are merged. We no longer need to manually delete in Azure first.

Fixes #204

### CODE REVIEW

File: .github/workflows/deployment.yaml [hunk 1]
```diff
@@ -220,25 +231,29 @@ jobs:
...
             if [[ $FILE == *.ipynb ]]; then
               FILENAME="$(basename $FILE .ipynb)"
-              if ! az synapse notebook show --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME; then
-                cat "../../scripts/Synapse/${FILENAME}.ipynb" | envsubst > "../../scripts/Synapse/${FILENAME}Substituted.ipynb"
-                az synapse notebook create --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME --file "@../../scripts/Synapse/${FILENAME}Substituted.ipynb" --spark-pool-name sparkpool
+              if az synapse notebook show --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME; then
+                az synapse notebook delete --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME --yes
               fi
+              cat "../../scripts/Synapse/${FILENAME}.ipynb" | envsubst > "../../scripts/Synapse/${FILENAME}Substituted.ipynb"
+              az synapse notebook create --workspace-name phdi${TF_ENV}synapse${SHORT_CID} --name $FILENAME --file "@../../scripts/Synapse/${FILENAME}Substituted.ipynb" --spark-pool-name sparkpool
```
Comment (m-goggins) [inline .github/workflows/deployment.yaml:238]: I wanted to double check that this deployment has run successfully. I don't think you can delete a notebook that is already linked up to a pipeline... or at least you cannot from the console; there are some diffs from the CLI that I may not be aware of. If it has worked, feel free to ignore.

If there are changes to the pipeline notebooks, one possible solution is to re-order the way things are deleted and created: 1) delete the pipeline trigger, 2) delete the pipeline, 3) delete the notebooks, 4) create the notebooks, 5) create the pipeline, and 6) create the trigger.

Comment (DanPaseltiner) [reply .github/workflows/deployment.yaml:238]: I have only run one test so far, but there were no changes to deploy so it tested a different case. This is a great point I'll investigate further.

**Review by bamader — APPROVED** [review]
Assuming the point Marcelle raised around pipeline triggers is okay, the rest of this looks good to me!

File: .github/workflows/deployment.yaml (hunk 1 above)
Comment (DanPaseltiner) [reply .github/workflows/deployment.yaml:238]: Just did some testing here and it appears that what @m-goggins has seen through the UI holds true with CLI. Unfortunately the `delete` command appears to succeed for notebooks attached to pipelines even though the notebook has not  truly been deleted. I'm going to modify things as your suggest.

**Comment by DanPaseltiner** [comment]
Modified following @m-goggins suggestions. [Successful deploy](https://github.com/CDCgov/phdi-azure/actions/runs/5522116199)

**Review by m-goggins — APPROVED** [review]
Looks good!

**Pull request merged**